### PR TITLE
Update main.py

### PR DESCRIPTION
**Pull request by martmists**
Bugfixes:
- Fix bug where if `default=True` in `ask_question` would cause `y` to return False

Improvements:
- Switch `not ... in ...` to `... not in ...`
- Reorder imports
- Remove places where builtins were shadowed (`list`, `id`)
- move `json.dumps/json.loads` to `json.dump/json.load` instead

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -69,33 +77,33 @@ def get_mod_info(url):
...
         'authors': owners,
         'name': soup.select('.project-title a span')[0].text,
-        'id': id,
+        'id': id_,
         'path_name': url.split('/projects/')[1]
     }
 
 
-def get_dl_info(url, version=VERSIONS.MC_1_12_2):
+def get_dl_info(url: str, version: str = VERSIONS.MC_1_12_2) -> dict:
```
[the remark this answers is not in the document]
Comment (martmists) [reply main.py]: nope, cause VERSIONS is an enum, not a type

File: main.py [hunk 2]
```diff
@@ -112,7 +120,7 @@ def get_dl_info(url, version=VERSIONS.MC_1_12_2):
     }
 
 
-def add_mod(url, version=VERSIONS.MC_1_12_2):
+def add_mod(url: str, version: str = VERSIONS.MC_1_12_2) -> None:
```
[the remark this answers is not in the document]
Comment (martmists) [reply main.py]: see above

File: main.py [hunk 3]
```diff
@@ -170,15 +178,14 @@ def add_mod(url, version=VERSIONS.MC_1_12_2):
         {'true' if len(dl_info['deps']) > 0 else 'false'}, // core,
         setOf() // dependencies
     )""")
+
     except NoDownloadsException:
-        return
+        print(f"No download for: {url:r}")
```
[the remark this answers is not in the document]
Comment (martmists) [reply main.py]: Do we want it to return silently and not tell the user why they cant access the owo server tho 🤔

File: main.py [hunk 4]
```diff
@@ -19,31 +27,31 @@
 class VERSIONS(enum.Enum):
     MC_1_12_2 = '2020709689%3A6756'
     MC_1_12_1 = '2020709689%3A6711'
-    MC_1_12 = '2020709689%3A6580'
+    MC_1_12   = '2020709689%3A6580'
```
[the remark this answers is not in the document]
Comment (martmists) [reply main.py]: fixed

File: main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (noud02) [reply main.py]: it isn't a string either

File: main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (noud02) [reply main.py]: If there aren't any dls it wouldn't be in the list

**Pull request merged**